pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -42,8 +42,14 @@ let
   # The wrapper scripts use 'cat' and 'grep', so we may need coreutils.
   coreutils_bin = if nativeTools then "" else getBin coreutils;
 
-  default_cxx_stdlib_compile = optionalString (targetPlatform.isLinux && !(cc.isGNU or false) && !nativeTools)
-    "-isystem $(echo -n ${cc.gcc}/include/c++/*) -isystem $(echo -n ${cc.gcc}/include/c++/*)/$(${cc.gcc}/bin/gcc -dumpmachine)";
+  default_cxx_stdlib_compile = if (targetPlatform.isLinux && !(cc.isGNU or false) && !nativeTools) then
+      "-isystem $(echo -n ${cc.gcc}/include/c++/*) -isystem $(echo -n ${cc.gcc}/include/c++/*)/$(${cc.gcc}/bin/gcc -dumpmachine)"
+    else if (targetPlatform.isDarwin && (cc.isClang or true) && !nativeTools && libcxx != null) then
+      "-isystem ${libcxx}/include/c++/v1"
```

Comment:
Should this be `${libcxx}/include/c++/*` instead like the gcc one? I'm not sure which is the correct way, especially since if it has multiple versions that would make the oldest version win, assuming the `*` works like shell globs.

Reply:
I might be wrong, but I think this has impact on bootstrapping the darwin stdenv.

Reply:
I personally would have preferred passing `libcxx` only from `all-packages.nix`'s `clang`. My interest is in having `nix-env -iA clang` install a working clang on darwin. I understand that when using clang from within nixpkgs the right flags are threaded through via the hooks and environment variables.

But it didn't seem like `nix-env -iA clang` was installing that clang at all, at least `builtins.trace` showed that wasn't getting evaluated at all. I'm a bit confused on how `nix-env -iA` finds the derivations to install.